### PR TITLE
Warnings removals and python 3.9 drop in tests

### PR DESCRIPTION
**Pull request by wpotrzebowski**
## Description
It is now down to one warning, which cannot easily be fixed due to dependent action. 

Removing warnings created by GH actions and dropping support for python 3.9 in tests (as it seems to be not available on `latest-macos` runners https://github.com/actions/setup-python/issues/850)

### CODE REVIEW

**Comment by rozyczko** [comment]
Just a note: how is this PR different from @butlerpd 's https://github.com/SasView/sasview/pull/2836 ?

**Comment by wpotrzebowski** [comment]
Ah, I thought I saw it somewhere already... @rozyczko Thanks for point it out. I will merge/move to #2836

File: .github/workflows/matrix.py [hunk 1]
```diff
@@ -43,8 +43,8 @@
 
 # List of python versions to use for tests
 python_test_list = python_release_list + [
-    '3.9',
-    '3.10'
+    '3.10',
+    '3.11'
```
[the remark this answers is not in the document]
Comment (wpotrzebowski) [reply .github/workflows/matrix.py:47]: @krzywon should we keep it for 3.10 only then?

[the remark this answers is not in the document]
Comment (krzywon) [reply .github/workflows/matrix.py:47]: At this point, I would say yes. We haven't tested 3.12 and introducing it this close to a release is probably a bad idea.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -278,14 +278,14 @@ jobs:
...
-        uses: actions/upload-artifact@v3
+        uses: actions/upload-artifact@v4
         with:
           name: SasView-Installer-${{ matrix.os }}-${{ matrix.python-version }}
           path: |
             installers/dist/setupSasView.exe
             installers/dist/SasView6.dmg
             installers/dist/sasview6.tar.gz
-          if-no-files-found: error
+          if-no-files-found: ignore
```
Comment (rozyczko) [inline .github/workflows/ci.yml:288]: Is this a good idea? Shouldn't the build be marked `failed` if the installer is not present?

**Review by rozyczko — APPROVED** [review]
Looks good. Works as expected.
One minor issue raised which does not block merging.

**Pull request merged**